test/datatable.jl

```diff
@@ -234,140 +232,141 @@ module TestDataTable
-                   Key = Nullable{String}["Mass", "Color", "Mass", "Color"],
-                   Value = Nullable{String}["12 g", "Red", "18 g", "Grey"])
+    dt = DataTable(Fish = CategoricalArray{Union{String, Null}}(["Bob", "Bob", "Batman", "Batman"]),
+                   Key = Union{String, Null}["Mass", "Color", "Mass", "Color"],
+                   Value = Union{String, Null}["12 g", "Red", "18 g", "Grey"])
     # Check that reordering levels does not confuse unstack
     levels!(dt[1], ["XXX", "Bob", "Batman"])
     #Unstack specifying a row column
-    dt2 = unstack(dt,:Fish, :Key, :Value)
+    dt2 = unstack(dt, :Fish, :Key, :Value)
     #Unstack without specifying a row column
-    dt3 = unstack(dt,:Key, :Value)
+    #FIXME: categoricalarrays
+    # dt3 = unstack(dt, :Key, :Value)
     #The expected output
-    dt4 = DataTable(Fish = Nullable{String}["XXX", "Bob", "Batman"],
-                    Color = Nullable{String}[Nullable(), "Red", "Grey"],
-                    Mass = Nullable{String}[Nullable(), "12 g", "18 g"])
-    @test isequal(dt2, dt4)
-    @test typeof(dt2[:Fish]) <: NullableCategoricalArray{String,1,UInt32}
+    dt4 = DataTable(Fish = Union{String, Null}["XXX", "Bob", "Batman"],
+                    Color = Union{String, Null}[null, "Red", "Grey"],
+                    Mass = Union{String, Null}[null, "12 g", "18 g"])
+    @test dt2 == dt4
+    @test typeof(dt2[:Fish]) <: CategoricalVector{Union{String, Null}}
     # first column stays as CategoricalArray in dt3
-    @test isequal(dt3[:, 2:3], dt4[2:3, 2:3])
+    # @test dt3[:, 2:3] == dt4[2:3, 2:3]
     #Make sure unstack works with NULLs at the start of the value column
-    dt[1,:Value] = Nullable()
-    dt2 = unstack(dt,:Fish, :Key, :Value)
+    dt[1,:Value] = null
+    dt2 = unstack(dt, :Fish, :Key, :Value)
     #This changes the expected result
-    dt4[2,:Mass] = Nullable()
-    @test isequal(dt2, dt4)
+    dt4[2,:Mass] = null
+    @test dt2 == dt4
 
     dt = DataTable(A = 1:10, B = 'A':'J')
     @test !(dt[:,:] === dt)
 
     @test append!(DataTable(A = 1:2, B = 1:2), DataTable(A = 3:4, B = 3:4)) == DataTable(A=1:4, B = 1:4)
-    dt = DataTable(A = NullableArray(1:3), B = NullableArray(4:6))
-    @test all(c -> isa(c, NullableArray), categorical!(deepcopy(dt)).columns)
-    @test all(c -> isa(c, NullableCategoricalArray), categorical!(deepcopy(dt), [1,2]).columns)
-    @test all(c -> isa(c, NullableCategoricalArray), categorical!(deepcopy(dt), [:A,:B]).columns)
-    @test find(c -> isa(c, NullableCategoricalArray), categorical!(deepcopy(dt), [:A]).columns) == [1]
-    @test find(c -> isa(c, NullableCategoricalArray), categorical!(deepcopy(dt), :A).columns) == [1]
-    @test find(c -> isa(c, NullableCategoricalArray), categorical!(deepcopy(dt), [1]).columns) == [1]
-    @test find(c -> isa(c, NullableCategoricalArray), categorical!(deepcopy(dt), 1).columns) == [1]
+    dt = DataTable(A = 1:3, B = 4:6)
```

Comment:
Need to test nullable vector. The element type of the `CategoricalArray` columns should also be checked.